index.js

```diff
@@ -112,21 +111,21 @@ function enterText(token) {
 }
 
 /**
- * @this {ThisParameterType<FromMarkdownHandle>}
- * @param {string} type
- * @param {Parameters<FromMarkdownHandle>[0]} token
+ * @this {CompileContext}
+ * @param {Directive['type']} type
+ * @param {Token} token
  */
 function enter(type, token) {
-  // @ts-expect-error: custom node.
   this.enter({type, name: '', attributes: {}, children: []}, token)
```

Comment:
This really feels like magic and is awesome: because `mdast-util-{to,from}-markdown` use `@types/mdast`, and this package adds Directive nodes, this just starts working!